File: DevExpress-Demo/CS/DockingDemo.Wpf/ViewModels/PinnedTabsViewModel.cs
using DockingDemo.Helpers;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Core;
using System.Linq;
using System.Windows.Media;
using System.Collections.ObjectModel;
using DevExpress.Xpf.Layout.Core;

namespace DockingDemo.ViewModels {
    public class PinnedTabsViewModel {
        public static PinnedTabsViewModel Create() {
            return ViewModelSource.Create(() => new PinnedTabsViewModel());
        }
        protected PinnedTabsViewModel() {
            Photos = new ObservableCollection<NaturePhoto>();
            for(int i = 0; i < 6; i++) {
                var photo = ImagesHelper.NaturePhotos.ElementAt(i);
                Photos.Add(NaturePhoto.Create("Nature Photo #" + i.ToString(), photo));
            }
            Photos[0].TogglePin();
            Photos[0].PinLocation = TabHeaderPinLocation.Far;
        }

        public virtual ObservableCollection<NaturePhoto> Photos { get; protected set; }
    }
    public class NaturePhoto {
        public static NaturePhoto Create(string title, string photo) {
            return ViewModelSource.Create(() => new NaturePhoto(title, photo));
        }
        protected NaturePhoto(string title, string photo) {
            Title = title;
            Photo = photo;
            Pinned = false;
        }
        public string Title { get; private set; }
        public string Photo { get; private set; }
        public bool Pinned { get; set; }
        public TabHeaderPinLocation PinLocation { get; set; }
        public void TogglePin() {
            Pinned = !Pinned;
        }
    }
}

File: DevExpress-Demo/VB/DockingDemo.Wpf/ViewModels/PinnedTabsViewModel.vb
Imports DockingDemo.Helpers
Imports DevExpress.Mvvm.POCO
Imports System.Linq
Imports System.Collections.ObjectModel
Imports DevExpress.Xpf.Layout.Core

Namespace DockingDemo.ViewModels

    Public Class PinnedTabsViewModel

        Public Shared Function Create() As PinnedTabsViewModel
            Return ViewModelSource.Create(Function() New PinnedTabsViewModel())
        End Function

        Protected Sub New()
            Photos = New ObservableCollection(Of NaturePhoto)()
            For i As Integer = 0 To 6 - 1
                Dim photo = ImagesHelper.NaturePhotos.ElementAt(i)
                Photos.Add(NaturePhoto.Create("Nature Photo #" & i.ToString(), photo))
            Next

            Photos(0).TogglePin()
            Photos(0).PinLocation = TabHeaderPinLocation.Far
        End Sub

        Public Overridable Property Photos As ObservableCollection(Of NaturePhoto)
    End Class

    Public Class NaturePhoto

        Private _Title As String, _Photo As String

        Public Shared Function Create(ByVal title As String, ByVal photo As String) As NaturePhoto
            Return ViewModelSource.Create(Function() New NaturePhoto(title, photo))
        End Function

        Protected Sub New(ByVal title As String, ByVal photo As String)
            Me.Title = title
            Me.Photo = photo
            Pinned = False
        End Sub

        Public Property Title As String
            Get
                Return _Title
            End Get

            Private Set(ByVal value As String)
                _Title = value
            End Set
        End Property

        Public Property Photo As String
            Get
                Return _Photo
            End Get

            Private Set(ByVal value As String)
                _Photo = value
            End Set
        End Property

        Public Property Pinned As Boolean

        Public Property PinLocation As TabHeaderPinLocation

        Public Sub TogglePin()
            Pinned = Not Pinned
        End Sub
    End Class
End Namespace
